trestle/core/commands/author/folders.py

```diff
@@ -150,46 +143,37 @@ def template_validate(
     ) -> int:
         """Validate that the template is acceptable markdown."""
         if not self.template_dir.is_dir():
-            logger.error(
+            raise TrestleError(
                 f'Template directory {self.rel_dir(self.template_dir)} for task {self.task_name} does not exist.'
             )
-            return CmdReturnCodes.COMMAND_ERROR.value
         # get list of files:
         template_files = self.template_dir.rglob('*')
 
         for template_file in template_files:
-            if not file_utils.is_local_and_visible(template_file):
-                continue
-            elif template_file.is_dir():
-                continue
-            elif template_file.suffix.lower() == '.md':
-                if not readme_validate and template_file.name == 'readme.md':
-                    logger.error('Template directory contains a readme.md file and readme validation is off.')
-                    return CmdReturnCodes.COMMAND_ERROR.value
-                try:
+            try:
+                if not file_utils.is_local_and_visible(template_file):
+                    continue
+                elif template_file.is_dir():
```

Comment:
surprised to see a fairly low level try/except here - but i guess it's ok?

Reply:
I think in this case it is okay, as it provides some specific information to the user, which only this function knows (i.e. which template file is faulty)